src/hooks/useSmartSearch.ts

```diff
@@ -0,0 +1,58 @@
+import { useState } from 'react';
+import { SmartSearchFilterWithId, ZetkinSmartSearchFilter } from 'types/smartSearch';
+
+type InitialFilters = ZetkinSmartSearchFilter[] | SmartSearchFilterWithId[]
+
+type UseSmartSearch = {
+    addFilter: (filter: ZetkinSmartSearchFilter) => void; // addSmartSearchFilter
+    deleteFilter: (id: number) => void; // removeSmartSearchFilter
+    editFilter: (id: number, newFilterValue: SmartSearchFilterWithId) => void; // editSmartSearchFilter
+    filters: ZetkinSmartSearchFilter[];
+    filtersWithIds: SmartSearchFilterWithId[];
+}
+
+const useSmartSearch = (initialFilters: InitialFilters = []): UseSmartSearch => {
+    const initialFiltersWithIds = initialFilters.map((filter, index) => ({ ...filter, id: index }));
```

Comment:
If we're always overwriting/setting the `id` here, why is `initialFilters` typed as `ZetkinSmartSearchFilter[] | ZetkinSmartSearchFilterWithId[]`?